### PR TITLE
add v8-canary builds to www setup

### PR DESCRIPTION
**Pull request by rvagg**
some chakracore stuff in there too that's come in from an upstream sync

### CODE REVIEW

File: setup/www/resources/config/nightly-builder-v8-canary.json.j2 [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "jenkinsToken": "{{ jenkins_token }}",
+  "jenkinsJobUrl": "https://ci-release.nodejs.org/job/iojs+release",
+  "jenkinsCrumbUrl": "https://ci-release.nodejs.org/crumbIssuer/api/json",
+  "githubAuthUser": "{{ github_auth_user }}",
+  "githubAuthToken": "{{ github_auth_token }}",
+  "githubOrg": "nodejs",
+  "githubRepo": "node-v8",
+  "githubScheme": "https://github.com/",
+}
```
Comment (gibfahn) [inline setup/www/resources/config/nightly-builder-v8-canary.json.j2]: missing `\n` at the end of the file it seems.

Comment (rvagg) [inline setup/www/resources/config/nightly-builder-v8-canary.json.j2]: ya, edited that one with sublime, shouldn't matter tho, it's still json

Comment (gibfahn) [inline setup/www/resources/config/nightly-builder-v8-canary.json.j2]: Sublime doesn't add newlines at the end of files? Huh...

AIUI the problem is that some tools (editors or git or something) require a newline at the end. It's never caused an issue for me so 🤷‍♂️ .

**Comment by joaocgreis** [comment]
For reference, the ChakraCore changes had been in https://github.com/nodejs/build/pull/547 , I didn't land them because I wanted to keep it more dynamic, but it's been there for a while and there's no harm in landing either.

**Comment by joaocgreis** [comment]
@rvagg looking at https://github.com/nodejs/build/pull/547/files , `user.yaml` has two entries: dist and staging directories. Isn't this PR missing the staging one?

File: setup/www/tasks/tools.yaml [hunk 2]
```diff
@@ -23,8 +23,8 @@
     dest: /etc/crontab
     line: "{{ item }}"
   with_items:
-    - '0 20    * * *   dist    /usr/bin/nodejs-nightly-builder --type nightly --ref heads/v6.x --config /etc/nightly-builder.json'
-    - '0 19    * * *   dist    /usr/bin/nodejs-nightly-builder --type nightly --ref heads/v7.x --config /etc/nightly-builder.json'
+    - '0 20    * * *   dist    /usr/bin/nodejs-nightly-builder --type v8-canary --ref heads/canary --config /etc/nightly-builder-v8-canary.json'
+    - '0 19    * * *   dist    /usr/bin/nodejs-nightly-builder --type nightly --ref heads/v8.x --config /etc/nightly-builder.json'
```
Comment (joaocgreis) [inline setup/www/tasks/tools.yaml]: Updated the branch to `heads/v8.x-staging` in the server, as per discussion in https://github.com/nodejs/LTS/issues/229 .

**Comment by rvagg** [comment]
fixed the problem in user.yml that @joaocgreis pointed out and merged it

**Pull request closed without merging**